### PR TITLE
DEPR: remove Index fastpath kwarg

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -259,14 +259,7 @@ def _outer_indexer(self, left, right):
...
     def __new__(
-        cls,
-        data=None,
-        dtype=None,
-        copy=False,
-        name=None,
-        fastpath=None,
-        tupleize_cols=True,
-        **kwargs,
+        cls, data=None, dtype=None, copy=False, name=None, tupleize_cols=True, **kwargs,
```
Comment (WillAyd) [inline pandas/core/indexes/base.py]: Not sure how we historically handle but unfortunate this accepts kwargs - I suppose if someone keeps supplying `fastpath` nothing will change. Not sure if we should explicitly raise now in that case

Comment (jbrockmendel) [reply pandas/core/indexes/base.py]: I changed `test_deprecated_fastpath` so instead of checking for a FutureWarning, it checks for a TypeError

**Comment by jbrockmendel** [comment]
@WillAyd are the mypy complaints actually caused by something here or just a coincidence?

**Comment by WillAyd** [comment]
Doesn't make sense on first glance; just restarted let's see

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -269,6 +269,8 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 
 **Other removals**
 
+- Removed the previously deprecated :meth:`Index.summary` (:issue:`18217`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: yes, this was what I had in mind for whatsnew placing.

**Comment by jbrockmendel** [comment]
@simonjayhawkins any ideas whats going on with mypy here?

**Comment by charlesdong1991** [comment]
might be relevant to this issue @jbrockmendel 

https://github.com/python/mypy/issues/1174

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins any ideas whats going on with mypy here?

we had this issue before, see https://github.com/pandas-dev/pandas/pull/29205#issuecomment-546030059

did reach an agreement whether it is a bug or expected behaviour.

the change to the function signatures has triggered this.

**Comment by jbrockmendel** [comment]
> did reach an agreement whether it is a bug or expected behaviour.

suggestions on what to do here?  It looks like @TomAugspurger was working on making a fixture for just [Index, Series] at one point

**Comment by TomAugspurger** [comment]
> fixture for just [Index, Series] at one point

Do you mean a TypeVar?

I don't recall the details, but I think my preference was to add `type: ignore` to the lines in the test files that were causing the (maybe buggy) inference by mypy. But I didn't really understand the issue enough to say what the expected / right behavior is.

**Review by jreback — APPROVED** [review]
lgtm. I see your typing TODOs, any objections to simply merging and fixing later? @simonjayhawkins

**Comment by simonjayhawkins** [comment]
> lgtm. I see your typing TODOs, any objections to simply merging and fixing later? @simonjayhawkins

sure

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

File: pandas/tests/arithmetic/test_numeric.py [hunk 3]
```diff
@@ -30,6 +31,19 @@ def adjust_negative_zero(zero, expected):
     return expected
 
 
+# TODO: remove this kludge once mypy stops giving false positives here
```
Comment (jbrockmendel) [inline pandas/tests/arithmetic/test_numeric.py:34]: @simonjayhawkins any idea if this is something we can address?

**Pull request merged**